epoxy-paging3/src/main/java/com/airbnb/epoxy/paging3/PagingDataEpoxyController.kt

```diff
@@ -0,0 +1,162 @@
+package com.airbnb.epoxy.paging3
+
+import android.annotation.SuppressLint
+import android.os.Handler
+import androidx.paging.CombinedLoadStates
+import androidx.paging.LoadState
+import androidx.paging.LoadType
+import androidx.paging.LoadType.REFRESH
+import androidx.paging.PagingData
+import androidx.paging.PagingSource
+import androidx.paging.RemoteMediator
+import androidx.recyclerview.widget.DiffUtil
+import com.airbnb.epoxy.EpoxyAsyncUtil
+import com.airbnb.epoxy.EpoxyController
+import com.airbnb.epoxy.EpoxyModel
+import com.airbnb.epoxy.EpoxyViewHolder
+import kotlinx.coroutines.ObsoleteCoroutinesApi
+
+/**
+ * An [EpoxyController] that can work with a [PagingData].
+ *
+ * Internally, it caches the model for each item in the [PagingData]. You must override
+ * [buildItemModel] method to build the model for the given item. Since [PagingData] might include
+ * `null` items if placeholders are enabled, this method needs to handle `null` values in the list.
+ *
+ * By default, the model for each item is added  to the model list. To change this behavior (to
+ * filter items or inject extra items), you can override [addModels] function and manually add built
+ * models.
+ *
+ * @param T The type of the item in the [PagingData].
+ */
+@ObsoleteCoroutinesApi
+abstract class PagingDataEpoxyController<T : Any>(
+    /**
+     * The handler to use for building models. By default this uses the main thread, but you can use
+     * [EpoxyAsyncUtil.getAsyncBackgroundHandler] to do model building in the background.
+     *
+     * The notify thread of your PagedList (from setNotifyExecutor in the pagingData Builder) must be
+     * the same as this thread. Otherwise Epoxy will crash.
+     */
+    modelBuildingHandler: Handler = defaultModelBuildingHandler,
+    /**
+     * The handler to use when calculating the diff between built model lists.
+     * By default this uses the main thread, but you can use
+     * [EpoxyAsyncUtil.getAsyncBackgroundHandler] to do diffing in the background.
+     */
+    diffingHandler: Handler = defaultDiffingHandler,
```

Comment:
since these handlers must use the same thread or crash it seems we should default them to the same handler